### DISCUSSION TITLE
Can PDFs be Repaired with PDFcpu?

### DISCUSSION THREAD

**Discussion by peterode** [pdfcpu/pdfcpu#757] [General]
I am seeking a command-line utility that can run on my Windows 10 system to use with a batch file.
I'd like to run many PDFs in many folders through a PDF repair utility and output to a new filename or subfolder.

I don't know which PDF files are corrupt or broken in some way, but some of my PDFs fail with other PDF utilities.
I've used the Stellar PDF Repair utility but don't like using the GUI, command-line will be faster.

I will run entire folders of PDFs through a repair utility, even good PDFs. Just to assure that my future interactions with ALL PDFs will be less troublesome.

I've never used Go, is there an executable file that I can run on Windows?
Or, is there a step-by-step instruction on how to build an executable (I can program with Smalltalk or C#, but not experienced with Go).
Thanks.

**Comment by hhrutter** [comment]
Hello & sorry for the delay..
pdfcpu does not offer a repair facility per se but you can try the `optimize command`:
`pdfcpu optimize in.pdf out.pdf`
